drivers/storage/portworx/component/px_resource_gateway.go

```diff
@@ -0,0 +1,444 @@
+		return err
+	}
+	if err := k8sutil.DeleteClusterRole(p.k8sClient, PxResourceGatewayClusterRoleName); err != nil {
+		return err
+	}
+	if err := k8sutil.DeleteClusterRoleBinding(p.k8sClient, PxResourceGatewayClusterRoleBindingName); err != nil {
+		return err
+	}
+	if err := k8sutil.DeleteDeployment(p.k8sClient, PxResourceGatewayDeploymentName, cluster.Namespace, *ownerRef); err != nil {
+		return err
+	}
+	p.MarkDeleted()
+	return nil
+}
+
+func (p *pxResourceGateway) MarkDeleted() {
+	p.isCreated = false
+}
+
+func (p *pxResourceGateway) createServiceAccount(
+	clusterNamespace string,
+	ownerRef *metav1.OwnerReference,
+) error {
+	return k8sutil.CreateOrUpdateServiceAccount(
+		p.k8sClient,
+		&v1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            PxResourceGatewayServiceAccountName,
+				Namespace:       clusterNamespace,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+		},
+		ownerRef,
+	)
+}
+
+func (p *pxResourceGateway) createClusterRole() error {
+	rules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"create", "get", "list", "patch", "update", "watch"},
+		},
+	}
+	return k8sutil.CreateOrUpdateClusterRole(
+		p.k8sClient,
+		&rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: PxResourceGatewayClusterRoleName,
+			},
+			Rules: rules,
+		},
+	)
+}
+
+func (p *pxResourceGateway) createClusterRoleBinding(
+	clusterNamespace string,
+) error {
+	return k8sutil.CreateOrUpdateClusterRoleBinding(
+		p.k8sClient,
+		&rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: PxResourceGatewayClusterRoleBindingName,
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      PxResourceGatewayServiceAccountName,
+					Namespace: clusterNamespace,
+				},
+			},
+			RoleRef: rbacv1.RoleRef{
+				Kind:     "ClusterRole",
+				Name:     PxResourceGatewayClusterRoleName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+	)
+}
+
+func (p *pxResourceGateway) createDeployment(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	imageName := p.getDesiredPxResourceGatewayImage(cluster)
+	if imageName == "" {
+		return fmt.Errorf("pxResourceGateway image cannot be empty")
+	}
+
+	existingDeployment := &appsv1.Deployment{}
+	err := p.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      PxResourceGatewayDeploymentName,
+			Namespace: cluster.Namespace,
+		},
+		existingDeployment,
+	)
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	targetCPU := defaultPxResourceGatewayCPU
+	if cpuStr, ok := cluster.Annotations[pxutil.AnnotationPxResourceGatewayCPU]; ok {
+		targetCPU = cpuStr
+	}
+	targetCPUQuantity, err := resource.ParseQuantity(targetCPU)
+	if err != nil {
+		return err
+	}
+
+	targetCPULimitQuantity, err := resource.ParseQuantity(defaultPxResourceGatewayCPULimit)
+	if err != nil {
+		return err
+	}
```

Comment:
Let's remove this. We can directly use the resources from the spec. We don't need the annotations. This was an old way that we used for some of the components.